### PR TITLE
229851 Upload Reviewer Step 4 - Return blob file for Ref requested

### PR DESCRIPTION
**Pull request by gldefra**

### CODE REVIEW

File: EST.MIT.InvoiceImporter.Function/Functions/ImporterFunctions.cs [hunk 1]
```diff
@@ -11,14 +11,14 @@
...
-    private readonly IBlobService _blobService;
+    private readonly IAzureBlobService _blobService;
     private readonly BlobServiceClient _blobServiceClient;
     private readonly IAzureTableService _azureTableService;
 
     private readonly IEventQueueService _eventQueueService;
     private readonly INotificationService _notificationService;
 
-    public ImporterFunctions(IBlobService blobService, IAzureBlobService azureBlobService, IAzureTableService azureTableService, IEventQueueService eventQueueService, INotificationService notificationService)
+    public ImporterFunctions(IAzureBlobService blobService, IAzureBlobService azureBlobService, IAzureTableService azureTableService, IEventQueueService eventQueueService, INotificationService notificationService)
```
Comment (andygibbonsdefra) [inline EST.MIT.InvoiceImporter.Function/Functions/ImporterFunctions.cs]: Do you need blobService AND azureBlobService params now? Both have same interface, will get injected with same service object?

Comment (gldefra) [reply EST.MIT.InvoiceImporter.Function/Functions/ImporterFunctions.cs]: good catch

**Review by andygibbonsdefra — DISMISSED** [review]

File: EST.MIT.InvoiceImporter.Function/Services/ServicesConfiguration.cs [hunk 2]
```diff
@@ -24,55 +25,56 @@ public static class ServicesConfiguration
...
-            }
-            else
+            services.AddSingleton<IEventQueueService>(_ =>
             {
-                return new EventQueueService(new QueueClient(configuration.GetSection("QueueConnectionString").Value, eventQueueName));
-            }
-        });
-       }
+                var eventQueueName = configuration.GetSection("EventQueueName").Value;
+                var queueConnectionString = configuration.GetSection("QueueConnectionString:Credential").Value;
```
Comment (jbarnsley10) [inline EST.MIT.InvoiceImporter.Function/Services/ServicesConfiguration.cs]: Confusing variable name - suggest 'queueStorageAccountCredential'

File: EST.MIT.InvoiceImporter.Function/Startup.cs [hunk 3]
```diff
@@ -39,9 +38,10 @@ public override void Configure(IFunctionsHostBuilder builder)
 
         builder.Services.AddTableBlobQueueServices(Configuration, mapperConfig);
 
+        builder.Services.AddSingleton<IAzureTableService, AzureTableService>();
```
Comment (jbarnsley10) [inline EST.MIT.InvoiceImporter.Function/Startup.cs]: Is AzureTableService not already defined in ServicesConfiguration?

File: EST.MIT.InvoiceImporter.Function/Startup.cs [hunk 4]
```diff
@@ -39,9 +38,10 @@ public override void Configure(IFunctionsHostBuilder builder)
...
         builder.Services.AddSingleton<IImporterFunctions, ImporterFunctions>();
         builder.Services.AddSingleton<IUploadFunctions, UploadFunctions>();
-        builder.Services.AddSingleton<IBlobService, BlobService>();
+        builder.Services.AddSingleton<IAzureBlobService, AzureBlobService>();
```
Comment (jbarnsley10) [inline EST.MIT.InvoiceImporter.Function/Startup.cs]: AzureBlobService is already defined in ServicesConfiguration

**Review by jbarnsley10 — CHANGES_REQUESTED** [review]
A couple of minor things. You've made it simpler by amalgamating BlobService and AzureBlobService - which is nice

**Review by jbarnsley10 — APPROVED** [review]

**Pull request merged**